### PR TITLE
Load providers dynamically

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

This is the first step to allow to separate providers from the rest of the code base.
So that it will be possible to have user created / custom providers, which will be maintained separately.
Also removed TNTvillage since it's dead and updated a few provider names.

### CODE REVIEW

File: medusa/providers/__init__.py [hunk 1]
```diff
@@ -3,78 +3,29 @@
...
+modules = [os.path.join(dirname, 'torrent', kind) for kind in torrent_kinds] + [os.path.join(dirname, 'nzb')]
+
+provider_names = []
+for (module_loader, name, ispkg) in pkgutil.iter_modules(modules):
+    base = os.path.basename(module_loader.path)
+    if base != 'nzb':
+        base = os.path.basename(os.path.dirname(module_loader.path)) + '.' + base
+    module = importlib.import_module('.' + name, __package__ + '.' + base)
+    if getattr(module, 'provider', None):
+        provider_names.append(module.provider.name.lower())
```
Comment (sharkykh) [inline medusa/providers/__init__.py]: Couldn't you use the filename of the provider file (module name) instead? Using the provider name is a bit risky if it contains non-ASCII characters.
Also this way you don't need to change provider names (that users see).

Comment (medariox) [reply medusa/providers/__init__.py]: Good point! I'm not even sure why I went for the provider name in the end... 🤔 The module name is definitely the better choice.

**Comment by p0psicles** [comment]
Nice PR guys! Keep it up!

**Comment by medariox** [comment]
Anyone wants to approve?

**Pull request merged**